text/header.tex

```diff
@@ -19,10 +19,13 @@ \section{The Header}\label{sec:header}
 
 We only require implementations to store headers of ancestors which were authored in the previous $\mathsf{L} = $ 24 hours of any block $\mathbf{B}$ they wish to validate.
 
-The extrinsic hash is the hash of the block's extrinsic data. Given any block $\mathbf{B} = (\mathbf{H}, \mathbf{E})$, then formally:
-\begin{equation}
-  \mathbf{H}_x \in \H \,,\quad \mathbf{H}_x \equiv \mathcal{H}(\mathcal{E}(\mathbf{E}))
-\end{equation}
+The extrinsic hash is a Merkle commitment to the block's extrinsic data, taking care to allow for the possibility of reports to individually have their inclusion proven. Given any block $\mathbf{B} = (\mathbf{H}, \mathbf{E})$, then formally:
+\begin{align}
+  \mathbf{H}_x &\in \H \ ,\quad
+  \mathbf{H}_x \equiv \mathcal{H}(\se(\mathcal{H}^\#(\mathbf{a}))) \\
+  \where \mathbf{a} &= [\se_T(\xttickets), \se_P(\xtpreimages), g, \se_A(\xtassurances), \se_D(\xtdisputes)] \\
+  \also g &= \se(\mathcal{H}^\#(\se^\#_g(\xtguarantees)))
```

Comment:
But this is a commitment to the *extrinsic*, not some components of the extrinsic. The extrinsic contains guarantees, not reports, and they're non-fungible.

Are you suggesting we instead have two-level hashing for guarantees in order that reports can be pre-distributed?

Reply:
The idea was to pre-distribute reports yes, as this needs to happen anyway to make the reports available to block builders. So when distributing a new block, instead of sending the full block, you send a "light" block with the work reports replaced by their hashes. With the extrinsic hash as you have defined it, it's not possible to verify that such a "light" block is correct without obtaining all the referenced work reports. If the extrinsic hash instead at some level contained the work report hashes then this would be possible. You could add an extra level of hashing, or you could move the guarantee hash "down" to just cover the work report, and include the rest of the guarantee bits inline in `g`.

Reply:
The idea was to pre-distribute guarantees, as this needs to happen anyway to make the guarantees available to block builders. When distributing a new block, instead of sending the full block, you send a "light" block with the _work reports_ replaced by their hashes. You then request any work reports you don't have using CE 136. You could of course have the "light" block replace things at the level of guarantees instead of work reports. I figured the work report level was better for the reasons given above. To allow this to work, you could add an extra level of hashing, or you could move the guarantee hash "down" to just cover the work report, and include the rest of the guarantee bits inline in `g`. Alternatively we could change SNP to have "light" blocks include guarantee hashes and add a separate protocol for requesting guarantees. I just don't want to do that without explicit agreement that this is the preferred way forward.

Reply:
ok - please see change